twoliter/src/project.rs

```diff
@@ -191,6 +210,38 @@ impl Project {
         modules.sort();
         Ok(modules)
     }
+
+    /// Returns a base64 encoded sha256 hash of the contents of the Project structure.
+    /// Used for verifying if a change would occure in Twoliter.lock
+    pub(crate) fn digest(&self) -> Result<String> {
```

Comment:
Does this function need to be updated anytime we change or add something to `Project` ? Or is it just going to be things that we care about in `.lock`?

Reply:
Primarily should be only the things we care about in .lock that would effect dependency changes, so if we make a change that we want to signal a need to regen Twoliter.lock we should change it